lms/djangoapps/grades/models.py

```diff
@@ -0,0 +1,253 @@
+"""
+Models used for robust grading.
+
+Robust grading allows student scores to be saved per-subsection independent
+of any changes that may occur to the course after the score is achieved.
+"""
+
+from django.db import models
+from django.db.utils import IntegrityError
+from model_utils.models import TimeStampedModel
+
+from coursewarehistoryextended.fields import UnsignedBigIntAutoField
+from xmodule_django.models import CourseKeyField, UsageKeyField
+
+from base64 import b64encode
+from collections import namedtuple
+from hashlib import sha256
+import json
+import logging
+
+
+log = logging.getLogger(__name__)
+
+
+BlockRecord = namedtuple('BlockRecord', ['locator', 'weight', 'max_score'])
+"""
+A namedtuple used to serialize information about a block at the time it was used in grade calculation.
+"""
```

Comment:
It looks like we have mixed practice across the codebase, but most common seems to be a comment before the namedtuple.  Since the docstring isn't actually a docstring (It doesn't get attached to the object's .__doc__ attribute when placed before or after the declaration), I've adopted this practice, and made this one a comment as well (before the declaration).